packages/node/src/node.ts

```diff
@@ -1 +1,99 @@
-export default class Node {}
+import * as cf from "@counterfactual/cf.js";
+import machine from "@counterfactual/machine";
+import * as ethers from "ethers";
+import EventEmitter from "eventemitter3";
+import firebase from "firebase";
+
+export default class Node {
+  // @ts-ignore
+  private instructionExecutor: machine.InstructionExecutor = Object.create(
+    null
+  );
+  private eventEmitter: EventEmitter;
+
+  constructor(
+    // @ts-ignore
+    private privateKey: string,
+    // @ts-ignore
+    private provider: ethers.providers.Web3Provider,
+    // @ts-ignore
+    private firestore: firebase.firestore.Firestore,
+    // @ts-ignore
+    private networkContext: Map<string, cf.types.Address>
+  ) {
+    this.eventEmitter = new EventEmitter();
+    this.registerListeners();
+  }
+
+  /**
+   * Create a multisig for the channel with the peer specified.
+   * @param peerAddress The peer to create the multisig with.
+   */
+  async createMultisig(
+    peerAddress: cf.types.Address
+  ): Promise<cf.types.Address> {
+    // TODO:
+    return Promise.resolve(ethers.constants.AddressZero);
+  }
+
+  /**
+   * Delegates setting up a listener to the Node's EventEmitter instance.
+   * @param event
+   * @param callback
+   */
+  on(event: string, callback: (res: cf.types.Node.MethodResponse) => void) {
+    this.eventEmitter.on(event, callback);
+  }
+
+  /**
+   * Delegates emitting to the Node's EventEmitter instance.
+   * @param event
+   * @param req
+   */
+  emit(event: string, req: cf.types.Node.MethodRequest) {
+    this.eventEmitter.emit(event, req);
+  }
+
+  /**
+   * This sets up all the listeners for the methods the Node is expected to have
+   * as described at https://github.com/counterfactual/monorepo/blob/master/packages/cf.js/API_REFERENCE.md#node-protocol
+   *
+   * The responses to these calls are the events being listened on
+   * https://github.com/counterfactual/monorepo/blob/master/packages/cf.js/API_REFERENCE.md#events
+   */
+  private registerListeners() {
+    this.eventEmitter.on(
+      cf.types.Node.MethodName.GET_APP_INSTANCES,
+      (req: cf.types.Node.MethodRequest) => {
+        // TODO: empty for now
+        const res: cf.types.Node.MethodResponse = {
+          type: req.type,
+          requestId: req.requestId,
+          result: this.getAppInstances()
+        };
+        this.eventEmitter.emit(req.requestId, res);
```

Comment:
Shouldn't this be `emit(res.type, res)`?

Reply:
Will update this as well.